src/fluree/db/connection.cljc

```diff
@@ -457,3 +458,30 @@
   [conn address data]
   (let [clg (-> conn :index-catalog :storage)]
     (storage/write-catalog-bytes clg address data)))
+
+(defn trigger-ledger-index
+  "Manually triggers indexing for a ledger/branch and waits for completion.
+   
+   Options:
+   - :branch - Branch name (defaults to main branch)
+   - :timeout - Max wait time in ms (default 300000 / 5 minutes)
+   
+   Returns the indexed database object or throws an exception on failure/timeout."
+  [conn ledger-alias opts]
+  (go-try
+    (let [{:keys [branch timeout]
+           :or {timeout 300000}} opts
+          ledger (<? (load-ledger-alias conn ledger-alias))
+          current-db (ledger/current-db ledger branch)
+          index-queue (-> ledger
+                          (ledger/get-branch-meta branch)
+                          :index-queue)
```

Comment:
I don't think we should be reaching into the index queue from here as that should be a hidden implementation detail that should change. I think we should add `trigger-index` apis to both the ledger and branch so we don't break the abstraction barrier.

Reply:
@zonotope addressed in 4817326feb0cf6f41fee869acdc647668c5e0795